### PR TITLE
Edit backup description

### PR DESCRIPTION
**Pull request by alexasparks**
Add ability to edit a backup description!

- Adds `editSnapshotMetaData` thunk which kicks off the `graphql` mutation flow & updates the snapshot description on hub
- Adds the `editSnapshotMetaData.fulfilled` & `editSnapshotMetaData.pending` extra reducers to the `snapshotsSlice` to update UI with updated description
- Shows an error banner if there's an issue updating the description

https://user-images.githubusercontent.com/10208759/127187999-148e72ce-9001-499e-b165-7e8a97099117.mov

### CODE REVIEW

**Review by tihal — CHANGES_REQUESTED** [review]
Just one possible change! Otherwise looking super slick!

**Review by veryspry — COMMENTED** [review]
Just for fun since I'm on paternity leave and miss the Local team :rofl:

File: src/renderer/components/modals/BackupEditDescriptionContents.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	site: Site;
+	snapshot: BackupSnapshot;
+}
+
+export const BackupEditDescriptionContents = (props: ModalContentsProps) => {
+	const [description, updateDescription] = useState('');
+	const { site, snapshot } = props;
+	const { configObject } = snapshot;
+
+	const updateDescriptionGQL = () => {
```
[the remark this answers is not in the document]
Comment (alexasparks) [reply src/renderer/components/modals/BackupEditDescriptionContents.tsx]: Heck yeah! I added it!

File: src/renderer/components/siteinfotools/SnapshotsTableList.tsx [hunk 2]
```diff
@@ -185,6 +200,16 @@ const LoadMoreWhenVisibleCell = ({ site }) => {
 	);
 };
 
+const renderDescription = (description?: string) => {
+	if (!description) {
+		return (
+			<LoadingIndicator style={{ margin: 0 }} color="Gray" />
+		);
+	}
```
[the remark this answers is not in the document]
Comment (alexasparks) [reply src/renderer/components/siteinfotools/SnapshotsTableList.tsx:208]: Nice catch! Totally overlooked that use case, should be fixed now!

**Review by tihal — APPROVED** [review]

**Pull request merged**